keps/976-plain-pods/README.md

```diff
@@ -0,0 +1,644 @@
+          count: 1,
+        },
+        {
+          name: workers,
+          count: 10,
+          template:
+            spec:
+              containers:
+                - name: worker
+                  requests:
+                    cpu: 1m
+        }
+      ]
+spec:
+  containers:
+  - name: job
+    image: hello-world
+    resources:
+      requests:
+        cpu: 1m
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: job-worker-1
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+    kueue.x-k8s.io/pod-group-name: pod-group
+    kueue.x-k8s.io/pod-group-role: worker
+spec:
+  containers:
+  - name: job
+    image: hello-world
+    resources:
+      requests:
+        cpu: 1m
+```
+
+### Notes/Constraints/Caveats (Optional)
+
+#### Skipping Pods belonging to queued objects
+
+Pods owned by jobs managed by Kueue should not be subject to extra management.
+These Pods can be identified based on the ownerReference. For these pods:
+- the webhook should not add a scheduling gate
+- the pod reconciler should not create a corresponding Workload object.
+
+#### Pods replaced on failure
+
+It is possible that users of plain Pods have a controller for them to handle failures and
+re-creations.
+
+Because Kueue can't know if Pods will be recreated or not, it will hold the entirety of the
+quota until it can determine that the whole Workload finished (all pods are terminated).
+In other words, Kueue won't support [dynamically reclaiming quota](https://github.com/kubernetes-sigs/kueue/issues/78)
+for plain Pods.
+
+### Risks and Mitigations
+
+#### Increased memory usage
+
+In order to support plain Pods, we need to start watching all Pods, even if they are not supposed
+to be managed by Kueue. This will increase the memory usage of Kueue just to maintain the
+informers.
+
+We can use the following mitigations:
+
+1. Drop the unused managedFields field from the Pod spec, like kube-scheduler is doing
+   https://github.com/kubernetes/kubernetes/pull/119556
+2. Filter out terminal Pods from informers, as they no longer influence quota usage
+   https://github.com/kubernetes/kubernetes/blob/99190634ab252604a4496882912ac328542d649d/pkg/scheduler/scheduler.go#L496
+
+## Design Details
+
+### Simplifying the Workload object
+
+When a Workload just represents a single Pod, it's wasteful to duplicate the pod spec on the
+Workload.
+Instead, the Workload podset can refer back to the Pod itself.
+
+Note that, for other objects, such as Job, the spec in the Workload serves as a snapshot of the
+original Job, so that it is possible to modify a Job during admission (to inject affinities) and
+revert the change on preemption. This is not a concern for Pods, because Pods can't be suspended.
+They terminate as Failed if preempted. As a result, there is no need for a snapshot of the
+original spec.
+
+The Workload's PodSet will look as follows:
+
+```golang
+type PodSet struct {
+  Name string
+  Count int32
+  Template *corev1.PodTemplateSpec
```

Comment:
> It would be breaking for users of the golang types, but not for the REST API.

It makes sense.

@alculquicondor As I can see in other sections, the `podRef` field still leaves. Can we clean up the `podRef` field from the whole of this KEP?

https://github.com/kubernetes-sigs/kueue/blob/d63751dee105caccd3c74f3a229f8fd88e7f4263/keps/976-plain-pods/README.md#single-pods